projects/plugins/jetpack/extensions/blocks/payments-intro/edit.js

```diff
@@ -1,20 +1,26 @@
 import { InnerBlocks, store as blockEditorStore } from '@wordpress/block-editor';
-import { createBlock, getBlockType, registerBlockVariation } from '@wordpress/blocks';
+import { cloneBlock, createBlock, getBlockType, registerBlockVariation } from '@wordpress/blocks';
 import { Placeholder } from '@wordpress/components';
 import { useDispatch, useSelect } from '@wordpress/data';
 import { useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import { get } from 'lodash';
 import PaymentsIntroBlockPicker from './block-picker';
+import PaymentsIntroPatternPicker from './pattern-picker';
 import defaultVariations from './variations';
 
 export default function JetpackPaymentsIntroEdit( { name, clientId, className } ) {
-	const { blockType, hasInnerBlocks } = useSelect( select => {
-		const { getBlocks } = select( blockEditorStore );
+	const patternFilter = pattern => {
+		return pattern.categories.includes( 'earn' );
+	};
```

Comment:
No need to change, I think we could do without a body.
```suggestion
	const patternFilter = pattern => pattern.categories.includes( 'earn' );
```

Reply:
This is not a big deal too, the pattern filter gets run twice, on line 23 and on line 65 it gets passed down all the way to the BlockPatternSetup component. Do you think we could do something about it?

Reply:
> the pattern filter gets run twice

`hasPatterns` is needed in this component so we can adjust the wording and decide to display the `PaymentsIntroPatternPicker`. The [upstream code](https://github.com/WordPress/gutenberg/blob/50f4ad8dcda8582e8a60419a6fb5c592348e5233/packages/block-editor/src/components/block-pattern-setup/index.js#L145) doesn't let us pass it a list of patterns, just a filter function.

The only thing we could do about it is to re-implement some of the upstream components ourselves and change them. This is overkill for the sake of searching an array of a few hundred entries.